### PR TITLE
reorder REST API doc

### PR DESCRIPTION
**Pull request by lumenCodes**
I am applying for Outreachy and I took this task from the spreadsheet.

I updated part of the docs and fixed broken links. 
This PR fixes part of [#41](https://github.com/jupyterhub/outreachy/issues/41).  

 I would love to add screenshots to some sections of the documentation but I can't find where the Rest API is hosted the Swagger link is returning a 404 error see screenshot below.
image

### CODE REVIEW

**Comment by minrk** [comment]
You can view your rendered docs by clicking the readthedocs link in the checks dropdown for each PR:

Screen Shot 2022-10-20 at 10 36 06

File: docs/source/reference/rest.md [hunk 1]
```diff
@@ -4,15 +4,19 @@
...
+- [where to learn more about JupyterHub's API](#learn-more-about-the-api)
+
+Before we discuss about JupyterHub's REST API, you can learn about [REST APIs here](https://en.wikipedia.org/wiki/Representational_state_transfer). A REST
+API provides a standard way for users to get and send information to the
+Hub.
 
 ## What you can do with the API
 
-Using the [JupyterHub REST API][], you can perform actions on the Hub,
+Using the [JupyterHub REST API](https://jupyterhub.readthedocs.io/en/stable/reference/rest-api.html#/default/delete_groups__name_), you can perform actions on the Hub,
```
Comment (minrk) [inline docs/source/reference/rest.md:19]: This already had a correct link

**Comment by lumenCodes** [comment]
> You can view your rendered docs by clicking the readthedocs link in the checks dropdown for each PR:
> 
> Screen Shot 2022-10-20 at 10 36 06

Okay, Thanks @minrk

File: docs/source/reference/rest.md [hunk 2]
```diff
@@ -4,15 +4,19 @@
 
 This section will give you information on:
 
-- what you can do with the API
-- create an API token
-- add API tokens to the config files
-- make an API request programmatically using the requests library
-- learn more about JupyterHub's API
+- [what you can do with the API](#what-you-can-do-with-the-api)
```
Comment (minrk) [inline docs/source/reference/rest.md:7]: I don't think in-page links like this are beneficial. There's already a table of contents links on the side bar that will always be correct, whereas these links will all break without warning if we change the spelling of headings.

File: docs/source/reference/rest.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ added to JupyterHub using `.api_tokens` configuration setting in
 `jupyterhub_config.py`.
 
 Alternatively, use the `jupyterhub token` command to generate a token
-for a specific hub user by passing the 'username':
+for a specific hub user by passing the " **username**":
```
Comment (minrk) [inline docs/source/reference/rest.md:44]: bold _or_ quotes, I wouldn't do both

File: docs/source/reference/rest.md [hunk 4]
```diff
@@ -123,9 +123,9 @@ c.JupyterHub.load_roles = [
 ```
 
 The token will have the permissions listed in the role
-(see [scopes][] for a list of available permissions),
+(see [scopes](https://jupyterhub.readthedocs.io/en/stable/rbac/scopes.html#scopes-in-jupyterhub) for a list of available permissions),
```
Comment (minrk) [inline docs/source/reference/rest.md:126]: This already has a link target, it's not missing

File: docs/source/reference/rest.md [hunk 5]
```diff
@@ -123,9 +123,9 @@ c.JupyterHub.load_roles = [
...
 but there will no longer be a user account created to house it.
-The main noticeable difference is that there will be no notebook server associated with the account
+The main noticeable difference between a configuration and a service is that there will be no notebook server associated with the account
```
Comment (minrk) [inline docs/source/reference/rest.md:128]:
```suggestion
The main noticeable difference between a user and a service is that there will be no notebook server associated with the account
```

File: docs/source/reference/rest.md [hunk 6]
```diff
@@ -136,8 +136,8 @@ Authorization header.
 ### Use requests
 
 Using the popular Python [requests](https://docs.python-requests.org)
-library, here's example code to make an API request for the users of a JupyterHub
-deployment. An API GET request is made, and the request sends an API token for
+library, (here's example code to make an API request for the users of a JupyterHub
+deployment) an API GET request is made, and the request sends an API token for
```
Comment (minrk) [inline docs/source/reference/rest.md:140]: I don't think a parenthetical helps clarity here. If we have a problem, maybe move the 'here's example...' bit to its own sentence at the end?

File: docs/source/reference/rest.md [hunk 7]
```diff
@@ -175,10 +175,10 @@ r.raise_for_status()
 r.json()
 ```
 
-The same API token can also authorize access to the [Jupyter Notebook REST API][]
-provided by notebook servers managed by JupyterHub if it has the necessary `access:users:servers` scope:
+The same API token can also authorize access to the [Jupyter Notebook REST API](https://jupyterhub.readthedocs.io/en/stable/reference/rest-api.html#/)
```
Comment (minrk) [inline docs/source/reference/rest.md:178]: As above, this already has a correct link target.

File: docs/source/reference/rest.md [hunk 8]
```diff
@@ -175,10 +175,10 @@ r.raise_for_status()
...
+provided by notebook servers managed by JupyterHub if it has the necessary `access:users:servers` scope.
 
-(api-pagination)=
+<p>&nbsp;</p>
```
Comment (minrk) [inline docs/source/reference/rest.md:181]: This is a MyST link target. Please don't remove them. In general, it's a good idea to. not remove things if you don't recognize them. THey are probably there for a reason, and you can ask what they are for.

File: docs/source/reference/rest.md [hunk 9]
```diff
@@ -303,7 +303,7 @@ or kubernetes pods.
 
 ## Learn more about the API
 
-You can see the full [JupyterHub REST API][] for details.
+You can see the full [JupyterHub REST API](https://jupyterhub.readthedocs.io/en/stable/reference/rest-api.html#/) for details.
```
Comment (minrk) [inline docs/source/reference/rest.md:306]: Same here: the link target is already correct.

**Review by minrk — COMMENTED** [review]
Thanks for your contribution! I’ve left some suggestions inline for you to incorporate. You can update this PR with the suggestions and we will come back to review again

**Comment by lumenCodes** [comment]
@minrk Please help me to review this I have applied the changes.

**Comment by minrk** [comment]
Great, thank you!

**Pull request merged**